src/game_interpreter.cpp

```diff
@@ -5377,54 +5422,124 @@ bool Game_Interpreter::CommandEasyRpgProcessJson(lcf::rpg::EventCommand const& c
 
-	if (operation == 0) { // Get operation: Extract a value from JSON data
+	switch (operation) {
+	case 0: { // Get operation: Extract a value from JSON data
 		result = Json_Helper::GetValue(*json_data, json_path);
-
 		if (result) {
-			switch (target_var_type) {
-			case 0: // Switch
-				Main_Data::game_switches->Set(target_var_id, atoi(result->c_str()) != 0);
-				break;
-			case 1: // Variable
-				Main_Data::game_variables->Set(target_var_id, atoi(result->c_str()));
-				break;
-			case 2: // String
-				Main_Data::game_strings->Asg({ target_var_id }, *result);
-				break;
-			default:
-				Output::Warning("CommandEasyRpgProcessJson: Unsupported target_var_type {}", operation);
-				return true;
+			set_var_value(target_var_type, target_var_id, *result);
+		}
+		break;
+	}
+	case 1: { // Set operation: Update JSON data with a new value
+		std::string new_value = get_var_value(target_var_type, target_var_id);
+		if (!new_value.empty()) {
+			result = Json_Helper::SetValue(*json_data, json_path, new_value);
+			if (result) {
+				Main_Data::game_strings->Asg({ source_var_id }, *result);
 			}
 		}
+		break;
 	}
-	else if (operation == 1) { // Set operation: Update JSON data with a new value
-		std::string new_value;
-
-		switch (target_var_type) {
-		case 0: // Switch
-			new_value = std::to_string(Main_Data::game_switches->Get(target_var_id));
-			break;
-		case 1: // Variable
-			new_value = std::to_string(Main_Data::game_variables->Get(target_var_id));
-			break;
-		case 2: // String
-			new_value = ToString(Main_Data::game_strings->Get(target_var_id));
-			break;
-		default:
-			Output::Warning("CommandEasyRpgProcessJson: Unsupported target_var_type {}", operation);
-			return true;
+	case 2: { // GetLength operation
+		auto length = Json_Helper::GetLength(*json_data, json_path);
+		if (length) {
+			std::string length_str;
+			if (target_var_type == 0) {
+				// For switches, true if length > 0
+				length_str = (*length > 0) ? "1" : "0";
+			}
+			else {
+				length_str = std::to_string(*length);
+			}
+			set_var_value(target_var_type, target_var_id, length_str);
 		}
-
-		result = Json_Helper::SetValue(*json_data, json_path, new_value);
-
-		if (result) {
-			Main_Data::game_strings->Asg({ source_var_id }, *result);
+		break;
+	}
+	case 3: { // GetKeys operation
+		auto keys = Json_Helper::GetKeys(*json_data, json_path);
+		if (keys) {
+			std::string keys_str;
+			for (size_t i = 0; i < keys->size(); ++i) {
+				if (i > 0) keys_str += ",";
+				keys_str += "\"" + (*keys)[i] + "\"";
+			}
+			std::string json_str = "{ \"keys\": [" + keys_str + "] }";
+			set_var_value(target_var_type, target_var_id, json_str);
+		}
+		break;
+	}
+	case 4: { // GetType operation
+		auto type = Json_Helper::GetType(*json_data, json_path);
+		if (type) {
+			std::string value;
+			if (target_var_type == 0) {
+				// For switches, true if it's an object or array
+				value = (*type == "object" || *type == "array") ? "1" : "0";
```

Comment:
This doesn't really make sense? Maybe just do not support Switch output for this or just use the same logic as for ``target_var_type == 1`` (always false, except when it exists)